DataFormats/CTPPSReco/interface/CTPPSPixelLocalTrack.h

```diff
@@ -0,0 +1,189 @@
+
+    CTPPSPixelLocalTrack() : z0_(0), chiSquared_(0), valid_(false), numberOfPointUsedForFit_(0)
+    {
+    }
+
+    CTPPSPixelLocalTrack(double z0, const TVectorD &track_params_vector,
+      const TMatrixD &par_covariance_matrix, double chiSquared);
+
+    virtual ~CTPPSPixelLocalTrack() {}
+
+    inline const edm::DetSetVector<CTPPSPixelFittedRecHit>& getHits() const { return track_hits_vector_; }
+    inline void addHit(unsigned int detId, const CTPPSPixelFittedRecHit &hit)
+    {
+      track_hits_vector_.find_or_insert(detId).push_back(hit);
+      if(hit.getIsUsedForFit()) ++numberOfPointUsedForFit_;
+    }
+
+    inline double getX0() const { return track_params_vector_[0]; }
+    inline double getX0Sigma() const { return sqrt(CovarianceMatrixElement(0, 0)); }
+    inline double getX0Variance() const { return CovarianceMatrixElement(0, 0); }
+
+    inline double getY0() const { return track_params_vector_[1]; }
+    inline double getY0Sigma() const { return sqrt(CovarianceMatrixElement(1, 1)); }
+    inline double getY0Variance() const { return CovarianceMatrixElement(1, 1); }
+
+    inline double getZ0() const { return z0_; }
+    inline void setZ0(double z0) { z0_ = z0; }
+
+    inline double getTx() const { return track_params_vector_[2]; }
+    inline double getTxSigma() const { return sqrt(CovarianceMatrixElement(2, 2)); }
+
+    inline double getTy() const { return track_params_vector_[3]; }
+    inline double getTySigma() const { return sqrt(CovarianceMatrixElement(3, 3)); }
+
+    inline TVector3 getDirectionVector() const
+    {
+      TVector3 vect(getTx(), getTy(), 1);
+      vect.SetMag(1.0);
+      return vect;
+    }
+
+    TVectorD getParameterVector() const;
+    void setParameterVector(const TVectorD & track_params_vector);
+
+    TMatrixD getCovarianceMatrix() const;
+    void setCovarianceMatrix(const TMatrixD &par_covariance_matrix);
+
+    inline double getChiSquared() const { return chiSquared_; }
+    inline void setChiSquared(double & chiSquared) { chiSquared_ = chiSquared; }
+
+    inline double getChiSquaredOverNDF() const { return chiSquared_ / (2*numberOfPointUsedForFit_ - 4); }
+
+    inline int getNDF() const {return (2*numberOfPointUsedForFit_ - 4); }
+
+    /// returns (x, y) vector
+    inline TVector2 getTrackPoint(double z) const 
+    {
+      double delta_z = z - z0_;
+      return TVector2(
+        track_params_vector_[0] + track_params_vector_[2] * delta_z,
+        track_params_vector_[1] + track_params_vector_[3] * delta_z);
+    }
+
+    inline TVector3 getTrackCentrePoint()
+    {
+      return TVector3(track_params_vector_[0], track_params_vector_[1], z0_);
+    }
+
+    TMatrixD trackPointInterpolationCovariance(double z) const;
+
+    inline bool isValid() const { return valid_; }
+
+    inline void setValid(bool valid) { valid_ = valid; }
+
+    friend bool operator< (const CTPPSPixelLocalTrack &l, const CTPPSPixelLocalTrack &r);
```

Comment:
perhaps better define inside the CTPPSPixelLocalTrack  ```bool operator < (const CTPPSPixelLocalTrack& a) const; ``` to keep the access to private members.

Reply:
BTW, where is this operator used?